flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/FromElementsFunction.java

```diff
@@ -62,6 +73,9 @@
 	/** Flag to make the source cancelable */
 	private volatile boolean isRunning = true;
 
+	private transient ListState<Integer> checkpointedState;
+
+	private static final String FROM_ELEMENT_STATE_NAME = "from-element-state";
```

Comment:
Instead of this you could also have
```
private static final ListStateDescriptor<Integer> STATE_DESCRIPTOR =
    new ListStateDescriptor<>("from-elements-state", IntSerializer.INSTANCE);
```

and then use this descriptor later directly instead of initialising with this field.

That's just a personal style nitpick. Your version is also fine. 😃